### PR TITLE
Swift: Emit diagnostics on assertion/expectation violations.

### PR DESCRIPTION
**Pull request by sashabu**
Closes https://github.com/github/codeql-team/issues/1992.

### CODE REVIEW

**Review by redsun82 — COMMENTED** [review]
I see the issue you ran into with `fmt::format` and RTTI.

I think long term (especially if we want to start using `fmt` in other context), it might actually be good to do a PR upstream. In particular I noticed that RTTI is only needed for `std::exception` formatting when the `t` option is provided, so the cleanest would be to make the `t` option an error when RTTI is disabled. But pushing the PR with tests and all might require some effort.

Short term another option is to provide bare-bone integrations ourselves. In order for formatting to be compatible with binlog, no formatting options are actually supported, which makes providing the formatting ourselves somewhat simpler maybe, because we do not require any type-specific option. So we could add
```cpp
template <>
struct fmt::formatter<std::filesystem::path>: fmt::formatter<std::string_view> {
  auto format(cosnt std::filesystem::path p, format_context& ctx) const {
    return fmt::formatter<std::string_view>::format(p.native(), ctx);
  }
};

template <>
struct fmt::formatter<std::error_code>: fmt::formatter<std::string_view> {
  auto format(std::error_code e, format_context& ctx) const {
    return fmt::formatter<std::string_view>::format(e.message(), ctx);
  }
};
```

File: swift/extractor/translators/DeclTranslator.cpp [hunk 1]
```diff
@@ -253,7 +253,7 @@ void DeclTranslator::fillFunction(const swift::AbstractFunctionDecl& decl,
   entry.name = !decl.hasName() ? "(unnamed function decl)" : constructName(decl.getName());
   entry.body = dispatcher.fetchOptionalLabel(decl.getBody());
   CODEQL_EXPECT_OR(return, decl.hasParameterList(), "Function {} has no parameter list",
-                         entry.name);
+                         entry.name.value_or("<unnamed>"));
```
[the remark this answers is not in the document]
Comment (sashabu) [reply swift/extractor/translators/DeclTranslator.cpp]: Done.

**Review by AlexDenisov — APPROVED** [review]

**Comment by AlexDenisov** [comment]
@sashabu could you please schedule a DCA run before merging?

**Comment by sashabu** [comment]
> could you please schedule a DCA run before merging?

@AlexDenisov Done. How does it look? It's my first time looking at a Swift run.

**Pull request merged**